libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -562,6 +589,20 @@ typeSpec ts = TypeSpec ts mempty
 
 -- The HasSpec Class ------------------------------------------------------
 
+-- | sizeOfSpec generalizes the method (sizeOfTypeSpec :: TypeSpec fn t -> SizeSpec) to (sizeOfSpec:: Spec fn t -> SizeSpec)
```

Comment:
Can we move this function elsewhere? It would be nice if the `The HasSpec Class` banner is above the `HasSpec` class implementation.